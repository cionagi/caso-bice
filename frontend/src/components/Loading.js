import React from 'react';

const Loading = () => {
  return (
    <div
      className="d-flex justify-content-center align-items-center"
      data-testid="loading-component"
    >
      <div className="spinner-border" style={{ width: '5rem', height: '5rem' }} role="status">
        <span className="sr-only">Loading...</span>
      </div>
    </div>
  );
};

export default Loading;
