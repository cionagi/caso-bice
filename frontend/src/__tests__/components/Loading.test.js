import React from 'react';
import { render } from '@testing-library/react';

import Loading from '../../components/Loading';

test('Link renders correctly', () => {
  const { asFragment, getByTestId } = render(<Loading />);
  expect(getByTestId('loading-component')).toHaveClass(
    ' d-flex justify-content-center align-items-center'
  );
  expect(asFragment()).toMatchSnapshot();
});
