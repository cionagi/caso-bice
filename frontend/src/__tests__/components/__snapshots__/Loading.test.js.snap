// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<DocumentFragment>
  <div
    class="d-flex justify-content-center align-items-center"
    data-testid="loading-component"
  >
    <div
      class="spinner-border"
      role="status"
      style="width: 5rem; height: 5rem;"
    >
      <span
        class="sr-only"
      >
        Loading...
      </span>
    </div>
  </div>
</DocumentFragment>
`;
